google-cloud-spanner/src/main/java/com/google/cloud/spanner/Options.java

```diff
@@ -1088,12 +1088,20 @@ void appendToOptions(Options options) {
 
     @Override
     public int hashCode() {
-      return RequestIdOption.class.hashCode();
+      return this.reqId.hashCode();
     }
 
     @Override
     public boolean equals(Object o) {
-      return o instanceof RequestIdOption;
+      // instanceof for a null object returns false.
+      if (!(o instanceof RequestIdOption)) {
+        return false;
+      }
+      RequestIdOption other = (RequestIdOption) o;
+      if (this.reqId == null || other.reqId == null) {
+        return this.reqId == null && other.reqId == null;
+      }
+      return this.reqId.equals(other.reqId);
```

Comment:
tip for next time (just leave as-is for now);Java has a utility method for this:

```java
java.util.Objects.equals(this.reqId, other.reqId);
```